vermouth/molecule.py

```diff
@@ -646,9 +664,14 @@ def edges_between(self, n_bunch1, n_bunch2):
             A list of tuples of edges in this molecule. The first element of
             the tuple will be in `n_bunch1`, the second element in `n_bunch2`.
         """
-        return [(node1, node2)
-                for node1, node2 in itertools.product(n_bunch1, n_bunch2)
-                if self.has_edge(node1, node2)]
+        result = []
+        set_1 = set(n_bunch1)
+        set_2 = set(n_bunch2)
+        for node1 in set_1:
+            cross = set_2 & set(self[node1])
+            for node2 in cross:
+                result.append((node1, node2))
+        return result
```

Comment:
Should this be an iterator? And possibly accept a `data=False` kwarg, which will make it yield `(n1, n2, data)`?

Reply:
The iterator is a good idea. for the `data` argument, do we need it? I'd like to avoid feature creeping the `Molecule` class.

Reply:
Now I see the point...

Reply:
I'm going to make your day worse :)
In NX, the data keyword takes either a bool and behaves as it does now, or an attribute key (and a default value?); in which case it doesn't give back the the whole dict, but just the requested element. I'm happy to keep this as is untill we need it though.

Reply:
:cry: It is doable. Though, the more feature we throw here, the slower the function will be; especially since it is within the loop.

Reply:
Let's just leave it as is for now. And remember to come back here once we need the functionality.